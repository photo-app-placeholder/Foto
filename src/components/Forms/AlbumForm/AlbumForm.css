.albumForm {
  display: flex;
  flex-direction: column;
  width: 200px;
  margin: auto;
}

.albumForm input {
  width: 100%;
  align-items: left;
  margin-bottom: 40px;
  margin-top: 20px;
}

.albumForm button {
  width: 100px;
  height: 20px;
}

.albumForm label {
  display: flex;
}

.check input {
  width: 20px;
  margin: 2px;
}

.disabled {
  background-color: rgba(75, 75, 75, 0.527);
  height: 30px;
  width: 200px;
}

.inputs {
  height: 30px;
  width: 200px;
}
