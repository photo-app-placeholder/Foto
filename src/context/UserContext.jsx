import { createContext, useContext, useState } from 'react';
import { getUser } from '../services/users';

export const UserContext = createContext();

export const UserProvider = ({ children }) => {
  const currentUser = getUser();
  const [user, setUser] = useState(
    currentUser ? { id: currentUser.id, email: currentUser.email } : {}
  );

  console.log(user);

  const contextValue = { user, setUser };

  return (
    <UserContext.Provider value={contextValue}>{children}</UserContext.Provider>
  );
};

export const useUser = () => {
  const context = useContext(UserContext);
  if (context === undefined) {
    throw new Error('Error: useUser needs to be used within UserContext');
  }
  return context;
};
