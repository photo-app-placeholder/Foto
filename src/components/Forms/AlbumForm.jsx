import React, { useState } from 'react';
import { useHistory } from 'react-router-dom';
import { useUser } from '../../context/UserContext';
import profileHook from '../../hooks/profileHook';
import { addAlbum } from '../../services/albums';

export default function AlbumForm() {
  const [title, setTitle] = useState('');
  const [privatePublic, setPrivatePublic] = useState(false);
  const [code, setCode] = useState('');
  const { user } = useUser();
  const history = useHistory();
  const { profile } = profileHook();
  console.log(user);

  const handleSubmit = async (e) => {
    try {
      e.preventDefault();
      await addAlbum({
        title: title,
        private_public: privatePublic,
        code: code,
        user_id: user.id,
      });
      alert(`${title} has been created.`);
      history.push(`/${profile.username}`);
    } catch {
      throw new Error('something went wrong creating album');
    }
  };

  return (
    <form>
      <input
        type="text"
        value={title}
        placeholder="Album Title"
        onChange={(e) => setTitle(e.target.value)}
      />
      <label>
        Lock album?
        <input
          type="checkbox"
          checked={privatePublic}
          onChange={(e) => setPrivatePublic(e.target.value)}
        />
      </label>
      {privatePublic ? (
        <label>
          Album Password:
          <input
            type="password"
            name="code"
            value={code}
            minLength="4"
            onChange={(e) => setCode(e.target.value)}
          />
        </label>
      ) : null}
      <button onClick={handleSubmit}>Make Album</button>
    </form>
  );
}
