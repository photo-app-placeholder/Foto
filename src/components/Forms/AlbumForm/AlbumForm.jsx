import React, { useState } from 'react';
import { useHistory } from 'react-router-dom';
import { useUser } from '../../../context/UserContext';
import profileHook from '../../../hooks/profileHook';
import { addAlbum } from '../../../services/albums';
import styles from './AlbumForm.css';

const { albumForm, check, disabled, inputs } = styles;

export default function AlbumForm() {
  const [title, setTitle] = useState('');
  const [privatePublic, setPrivatePublic] = useState(false);
  const [code, setCode] = useState('');
  const { user } = useUser();
  const history = useHistory();
  const { profile } = profileHook();
  const { username } = profile[0];

  const handleSubmit = async (e) => {
    try {
      e.preventDefault();
      await addAlbum({
        title: title,
        private_public: privatePublic,
        code: code,
        user_id: user.id,
      });
      alert(`${title} has been created.`);
      history.push(`/${username}`);
    } catch {
      throw new Error('something went wrong creating album');
    }
  };

  return (
    <div className={albumForm}>
      <form>
        <input
          className={inputs}
          type="text"
          value={title}
          placeholder="Album Title"
          onChange={(e) => setTitle(e.target.value)}
        />
        {}
        <div className={check}>
          <label>
            Make Private
            <input
              type="checkbox"
              checked={privatePublic}
              onClick={() => setPrivatePublic(!privatePublic)}
            />
          </label>
        </div>
        {privatePublic ? (
          <input
            className={inputs}
            type="password"
            name="code"
            placeholder="Access Code"
            minLength="4"
            onChange={(e) => setCode(e.target.value)}
          />
        ) : (
          <input
            className={disabled}
            type="password"
            name="code"
            value={code}
            placeholder="Access Code"
            minLength="4"
            readOnly
          />
        )}
        <button onClick={handleSubmit}>Create</button>
      </form>
    </div>
  );
}
