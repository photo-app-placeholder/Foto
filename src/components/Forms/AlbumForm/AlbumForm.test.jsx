import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { UserProvider } from '../../../context/UserContext';
import AlbumForm from './AlbumForm';

test('should display a form to add images', () => {
  const { container } = render(
    <UserProvider>
      <MemoryRouter>
        <AlbumForm />
      </MemoryRouter>
    </UserProvider>
  );
  expect(container).toMatchSnapshot();
});
