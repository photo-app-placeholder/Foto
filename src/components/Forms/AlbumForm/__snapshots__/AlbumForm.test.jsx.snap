// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display a form to add images 1`] = `
<div>
  <div
    class="albumForm"
  >
    <form>
      <input
        class="inputs"
        placeholder="Album Title"
        type="text"
        value=""
      />
      <div
        class="check"
      >
        <label>
          Make Private
          <input
            type="checkbox"
          />
        </label>
      </div>
      <input
        class="disabled"
        minlength="4"
        name="code"
        placeholder="Access Code"
        readonly=""
        type="password"
        value=""
      />
      <button>
        Create
      </button>
    </form>
  </div>
</div>
`;
